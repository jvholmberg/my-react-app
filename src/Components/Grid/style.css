.Grid {
  display: block;
  margin: 0 auto;
  float: left;
}

/* Extra small devices (portrait phones, less than 576px) */
@media (max-width: 575px) {
  .Grid {
    padding: 0 3rem 3rem ;
    float: none;
    box-sizing: border-box;
  }
  .Grid-Col12 {
    width: 100%;
  }
  .Grid-Col9 {
    width: 100%;
  }
  .Grid-Col8 {
    width: 100%;
  }
  .Grid-Col6 {
    width: 100%;
  }
  .Grid-Col4 {
    width: 100%;
  }
  .Grid-Col3 {
    width: 100%;
  }
}

/* Small devices (landscape phones, 576px and up) */
@media (min-width: 576px) and (max-width: 767px) {
  .Grid {
    padding: 0 5rem 3rem;
    box-sizing: border-box;
  }
  .Grid-Col12 {
    width: 100%;
  }
  .Grid-Col9 {
    width: 100%;
  }
  .Grid-Col8 {
    width: 100%;
  }
  .Grid-Col6 {
    width: 100%;
  }
  .Grid-Col4 {
    width: 100%;
  }
  .Grid-Col3 {
    width: 100%;
  }
}

/* Medium devices (tablets, 768px and up) */
@media (min-width: 768px) and (max-width: 991px) {
  .Grid {
    padding: 0 2rem 0;
    box-sizing: border-box;
  }
  .Grid-Col12 {
    width: 100%;
  }
  .Grid-Col9 {
    width: 75%;
  }
  .Grid-Col8 {
    width: 66.66%;
  }
  .Grid-Col6 {
    width: 50%;
  }
  .Grid-Col4 {
    width: 33.33%;
  }
  .Grid-Col3 {
    width: 25%;
  }
}

/* Large devices (desktops, 992px and up) */
@media (min-width: 992px) and (max-width: 1199px) {
  .Grid {
    padding: 0 3rem 0;
    box-sizing: border-box;
  }
  .Grid-Col12 {
    width: 100%;
  }
  .Grid-Col9 {
    width: 75%;
  }
  .Grid-Col8 {
    width: 66.66%;
  }
  .Grid-Col6 {
    width: 50%;
  }
  .Grid-Col4 {
    width: 33.33%;
  }
  .Grid-Col3 {
    width: 25%;
  }
}

/* Extra large devices (large desktops, 1200px and up) */
@media (min-width: 1200px) {
  .Grid {
    padding: 0 3rem 0;
    box-sizing: border-box;
  }
  .Grid-Col12 {
    width: 100%;
  }
  .Grid-Col9 {
    width: 75%;
  }
  .Grid-Col8 {
    width: 66.66%;
  }
  .Grid-Col6 {
    width: 50%;
  }
  .Grid-Col4 {
    width: 33.33%;
  }
  .Grid-Col3 {
    width: 25%;
  }
}
