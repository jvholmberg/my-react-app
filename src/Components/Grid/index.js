import React from 'react';
import { CraftClassName } from '../'

import './style.css';

const Grid = ({
  col12, col9, col8, col6, col4, col3, className, children, ...props,
}) => {
  let cn = CraftClassName({ col12, col9, col8, col6, col4, col3 }, 'Grid', 'Col12', 'Grid-');
  if (className) cn = `${className} ${cn}`;
  return (<div className={cn} {...props}>{children}</div>);
};
export default Grid;
