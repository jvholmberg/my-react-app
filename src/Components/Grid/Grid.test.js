import React from 'react';
import renderer from 'react-test-renderer';

import Grid from './';

describe('Grid', () => {
  it('Empty matches snapshot', () => {
    expect(shallow(
      <Grid />
    )).toMatchSnapshot();
  });
  it('With className matches snapshot', () => {
    expect(shallow(
      <Grid className='className' />
    )).toMatchSnapshot();
  });
});
