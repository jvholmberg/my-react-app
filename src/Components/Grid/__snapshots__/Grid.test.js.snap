// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Grid Empty matches snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Grid />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": undefined,
      "className": "Grid Grid-Col12",
    },
    "ref": null,
    "rendered": null,
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": undefined,
        "className": "Grid Grid-Col12",
      },
      "ref": null,
      "rendered": null,
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Grid With className matches snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Grid
    className="className"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": undefined,
      "className": "className Grid Grid-Col12",
    },
    "ref": null,
    "rendered": null,
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": undefined,
        "className": "className Grid Grid-Col12",
      },
      "ref": null,
      "rendered": null,
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
